src/vmm/src/lib.rs

```diff
@@ -162,27 +165,66 @@ impl TryFrom<VMMConfig> for VMM {
         let vm = KvmVm::new(&kvm, vm_state, &guest_memory)?;
 
         let mut vmm = VMM {
-            vm,
             kvm,
+            vm,
+            kernel_cfg: config.kernel_config.clone(),
             guest_memory,
             device_mgr: Arc::new(Mutex::new(IoManager::new())),
             event_mgr: EventManager::new().map_err(Error::EventManager)?,
-            kernel_cfg: config.kernel_config.clone(),
             block_devices: Vec::new(),
+            net_devices: Vec::new(),
         };
 
         vmm.create_vcpus(&config.vcpu_config)?;
         vmm.add_serial_console()?;
 
+        // Adding the virtio devices. We'll come up with a cleaner abstraction for `Env`.
+
+        let mem = Arc::new(vmm.guest_memory.clone());
+
         // We transiently define a mut `Cmdline` object here to pass for device creation
         // (devices expect a `&mut Cmdline` to leverage the newly added virtio_mmio
         // functionality), until we figure out how this fits with the rest of the vmm, which
         // apparently does not explicitly use `Cmdline` structs. Will discuss and fix this
```

Comment:
It would be nice if we could use `Cmdline` everywhere (hopefully, we will avoid this way adding spaces in strings like [here](https://github.com/rust-vmm/vmm-reference/blob/master/src/vmm/src/lib.rs#L312), but didn't look very much into how is `Cmdline` implemented).